src/auto-evo/RunResults.cs

```diff
@@ -811,6 +826,144 @@ void OutputPopulationForPatch(Species species, Patch patch, long population)
+                                patch.LogEvent(new LocalizedString("TIMELINE_SPECIES_POPULATION_DECREASE",
+                                        species.FormattedName, ((double)finalPatchPopulation).FormatNumber()),
+                                    species.PlayerSpecies, "popDown.png");
+                            }
+                        }
+                        else
+                        {
+                            patch.LogEvent(new LocalizedString(
+                                    "TIMELINE_SPECIES_EXTINCT_LOCAL", species.FormattedName),
+                                species.PlayerSpecies, "extinctionLocal.png");
+                        }
+
+                        if (globalPopulation > previousGlobalPopulation)
+                        {
+                            world.LogEvent(new LocalizedString("TIMELINE_SPECIES_POPULATION_INCREASE",
+                                    species.FormattedName, ((double)globalPopulation).FormatNumber()),
+                                species.PlayerSpecies, "popUp.png");
+                        }
+                        else
+                        {
+                            world.LogEvent(new LocalizedString("TIMELINE_SPECIES_POPULATION_DECREASE",
+                                    species.FormattedName, ((double)globalPopulation).FormatNumber()),
+                                species.PlayerSpecies, "popDown.png");
+                        }
+                    }
+                }
+
+                foreach (var migration in GetMigrationsTo(patch))
+                {
+                    // Log to destination patch
+                    patch.LogEvent(new LocalizedString("TIMELINE_SPECIES_MIGRATED_FROM", migration.Key.FormattedName,
+                            TranslationServer.Translate(migration.Value.From.Name)),
+                        migration.Key.PlayerSpecies, "newSpecies.png");
+
+                    // Log to game world
+                    world.LogEvent(new LocalizedString("GLOBAL_TIMELINE_SPECIES_MIGRATED_TO",
+                            migration.Key.FormattedName,
+                            TranslationServer.Translate(migration.Value.To.Name),
+                            TranslationServer.Translate(migration.Value.From.Name)),
+                        migration.Key.PlayerSpecies, "newSpecies.png");
+
+                    // Log to origin patch
+                    migration.Value.From.LogEvent(new LocalizedString("TIMELINE_SPECIES_MIGRATED_TO",
+                            migration.Key.FormattedName, TranslationServer.Translate(migration.Value.To.Name)),
+                        migration.Key.PlayerSpecies, "newSpecies.png");
+                }
+
+                foreach (var newSpeciesEntry in newSpecies)
+                {
+                    GetSpeciesPopulationsByPatch(newSpeciesEntry, true, true).TryGetValue(patch, out var population);
+
+                    var newSpeciesResult = GetNewSpeciesResult(newSpeciesEntry);
```

Comment:
Isn't the new species result data available through the newSpeciesEntry?

Reply:
It only provides `Species` data and not the species type result that we want.

Reply:
I think it would be better to have a method to go from Species to the SpeciesResult object, that way it's general and helpful to (future) code in this class.

Reply:
In that case, methods might be redundant, could probably just use the `results` dicitionary.

Reply:
Yeah, that's what the apply results does. I assumed there was *some* reason why you didn't do that here, so I didn't suggest it...

Reply:
The methods were just left-overs from previous implementation done outside RunResults which is why there are public methods. I'll remove them in favor in direct accessing.